### ISSUE TITLE
[META] Java 11 Support

### ISSUE THREAD

**Issue by andrewvc** [elastic/logstash#9345]
This issue tracks Java 10 support for Logstash.

Current blockers:
- [ ] Bundler works with JRuby 9.1.17.0+
- [x] https://github.com/jruby/jruby-openssl/issues/157 needs to be fixed in JRuby
- [ ] https://github.com/elastic/logstash/issues/9316 our GC settings will no longer work. 

We're currently in a holding pattern for a new JRuby release.

**Comment by AlexaSHOB** [comment]
HI Andrew, 
I have the same issue.
command : logstash -f logstash.conf
Unrecognized VM option 'UseParNewGC'
Error: Could not create the Java Virtual Machine.
Error: A fatal exception has occurred. Program will exit.

OS: Windows 10
JDK Version: 10

Are there any news regarding the fix of this issue? :-(

**Comment by andrewvc** [comment]
Hmmm, looks like the new JRuby version may work with Java 10, but it totally breaks our usage of bundler :-/ I'll add it to the checklist.

**Comment by original-brownbear** [comment]
@andrewvc bundler isn't the issue we have a working upgrade from older bundler. Right now JRuby 9.2 release is broken for our tests by https://github.com/jruby/jruby/issues/5193.
Even fixing that comes with 50+ test failures that we'll have to track down one by one once we have a JRuby release that is at least capable of executing tests.
Unfortunately there isn't that much point in testing against current `master` of JRuby because that's a moving target I'm afraid and we gotta fix and validate a JRuby release version for our use.

**Comment by JoshDaone** [comment]
Same problem with new package in Ubuntu:

root@pndesktopa:~# apt install logstash
Reading package lists... Done
Building dependency tree       
Reading state information... Done
The following NEW packages will be installed:
  logstash
0 upgraded, 1 newly installed, 0 to remove and 7 not upgraded.
Need to get 149 MB of archives.
After this operation, 249 MB of additional disk space will be used.
Get:1 https://artifacts.elastic.co/packages/6.x/apt stable/main amd64 logstash all 1:6.3.0-1 [149 MB]
Fetched 149 MB in 26s (5,750 kB/s)                                                                                      
Selecting previously unselected package logstash.
(Reading database ... 161933 files and directories currently installed.)
Preparing to unpack .../logstash_1%3a6.3.0-1_all.deb ...
Unpacking logstash (1:6.3.0-1) ...
Setting up logstash (1:6.3.0-1) ...
Using provided startup.options file: /etc/logstash/startup.options
Unrecognized VM option 'UseParNewGC'
Error: Could not create the Java Virtual Machine.
Error: A fatal exception has occurred. Program will exit.
Unable to install system startup script for Logstash.
chmod: cannot access '/etc/default/logstash': No such file or directory
dpkg: error processing package logstash (--configure):
 installed logstash package post-installation script subprocess returned error exit status 1
E: Sub-process /usr/bin/dpkg returned an error code (1)
root@pndesktopa:~# java -version
openjdk version "10.0.1" 2018-04-17
OpenJDK Runtime Environment (build 10.0.1+10-Ubuntu-3ubuntu1)
OpenJDK 64-Bit Server VM (build 10.0.1+10-Ubuntu-3ubuntu1, mixed mode)


root@pndesktopa:~# echo $JAVA_HOME
/usr/lib/jvm/java-11-openjdk-amd64

**Comment by vasiqmz** [comment]
I'm facing the same issue on Ubuntu Server with Java 10
Is there any temp fix for the issue, I read on few posts that if you switch to older Java version, it may help.

**Comment by original-brownbear** [comment]
@vasiqmz currently there is no temporary fix sorry. Switching to Java 8 is the only stable solution for the time being.

**Comment by JoshDaone** [comment]
@original-brownbear thanks! Switched back to openjdk-8-jre and working properly.

**Comment by OzWookiee** [comment]
Issue has been open since April and is still waiting a fix. With the recent announcement of Oracle going to require paid licence for older Java (read Java-8) then we really need to get this fixed ASAP.

**Comment by ajaybhatnagar** [comment]
Tried with ELK stack 6.4.0. Issue continues. Environment : OS Ubuntu 4.4.0-134-generic. Java version "10.0.2" 2018-07-17.

root@elk00:/var/log# apt-get install logstash
Reading package lists... Done
Building dependency tree
Reading state information... Done
The following NEW packages will be installed:
  logstash
0 upgraded, 1 newly installed, 0 to remove and 0 not upgraded.
Need to get 0 B/156 MB of archives.
After this operation, 259 MB of additional disk space will be used.
Selecting previously unselected package logstash.
(Reading database ... 131521 files and directories currently installed.)
Preparing to unpack .../logstash_1%3a6.4.0-1_all.deb ...
Unpacking logstash (1:6.4.0-1) ...
Setting up logstash (1:6.4.0-1) ...
Using provided startup.options file: /etc/logstash/startup.options
**Unrecognized VM option 'UseParNewGC'**
Error: Could not create the Java Virtual Machine.
Error: A fatal exception has occurred. Program will exit.
chmod: cannot access '/etc/default/logstash': No such file or directory
dpkg: error processing package logstash (--configure):
 subprocess installed post-installation script returned error exit status 1
Errors were encountered while processing:
 logstash
E: Sub-process /usr/bin/dpkg returned an error code (1)

**Comment by reubensml** [comment]
Issue still exists

logstash -f logstash-syslog.conf
Unrecognized VM option 'UseParNewGC'
Error: Could not create the Java Virtual Machine.
Error: A fatal exception has occurred. Program will exit.
logstash-6.4.0l$ java --version
java 10.0.2 2018-07-17
Java(TM) SE Runtime Environment 18.3 (build 10.0.2+13)
Java HotSpot(TM) 64-Bit Server VM 18.3 (build 10.0.2+13, mixed mode)

**Comment by guyboertje** [comment]
Please do not keep adding "it still does not work..." comments. We know and we have not forgotten.

When we release a version of LS that supports a newer version of Java than v8, we will close this issue and the release notes will say so.

We can't just make it work or fix it ASAP because we are waiting on fixes to versions of JRuby that do support Java 10/11 - the latest version of JRuby (as of 21 Jun) has bugs that affect Logstash badly.

So we could give you Java 10/11 support but with known bugs in Logstash but we are not going to do that.

**Comment by yaauie** [comment]
This has been superseded by #10279, which has been merged to `master` and backported to `6.x`; it is on track for upcoming releases:

| branch | pull-request | next release |
|-----|-----|-----|
| `6.x` | #10382 | `v6.7.0` |
| `master` | #10279 | `v7.0.0-beta1` |
